calendar/experiments/calendar/ext-calendar-utils.jsm

```diff
@@ -90,6 +95,26 @@ function convertCalendar(extension, calendar) {
   return props;
 }
 
+function getOccurrenceDate(item, occurrence) {
+  if (!occurrence?.occurrenceId && !occurrence?.occurrenceIndex) {
+    return null;
+  }
+  if (!item.recurrenceInfo) {
+    throw new ExtensionError("Must be recurring item");
+  }
+  if (occurrence.occurrenceId) {
+    let date = cal.createDateTime(occurrence.occurrenceId).getInTimezone(item.recurrenceStartDate.timezone);
+    date.isDate = item.recurrenceStartDate.isDate;
+    return date;
+  }
+  let [[rule]] = splitRecurrenceRules(item.recurrenceInfo);
+  let dates = rule.QueryInterface(Ci.calIRecurrenceRule).getOccurrences(item.recurrenceStartDate, item.recurrenceStartDate, null, occurrence.occurrenceIndex);
```

Comment:
@kewisch From time to time we see Owl users hit an `NS_NOINTERFACE` exception on code like this where `splitRecurrenceRules` doesn't actually return something that's a `calIRecurrenceRule`. Am I actually splitting the rules correctly? I thought that's how Lightning did it itself but in that case I don't understand how Owl users are getting errors.